frontend/awx/access/credentials/CredentialForm.tsx

```diff
@@ -284,25 +357,58 @@ function CredentialInputs({
   setCredentialPluginValues,
   accumulatedPluginValues,
   setAccumulatedPluginValues,
+  setSelectedCredentialTypeId,
+  setIsTestButtonEnabled,
+  setIsTestButtonEnabledSubForm,
+  setWatchedSubFormFields,
 }: {
   isEditMode?: boolean;
   selectedCredentialTypeId?: number;
   credentialTypes: CredentialTypes;
   setCredentialPluginValues?: (values: CredentialPluginsInputSource[]) => void;
   accumulatedPluginValues?: CredentialPluginsInputSource[];
   setAccumulatedPluginValues?: (values: CredentialPluginsInputSource[]) => void;
+  setSelectedCredentialTypeId?: (id: number) => void;
+  setIsTestButtonEnabled: (enabled: boolean) => void;
+  setIsTestButtonEnabledSubForm: (enabled: boolean) => void;
+  setWatchedSubFormFields: (fields: unknown[]) => void;
 }) {
   const { t } = useTranslation();
 
   const watchedCredentialTypeId = useWatch<{ credential_type: number }>({
     name: 'credential_type',
   });
 
+  useEffect(() => {
+    if (setSelectedCredentialTypeId) {
+      setSelectedCredentialTypeId(watchedCredentialTypeId);
+    }
+  }, [watchedCredentialTypeId, setSelectedCredentialTypeId]);
+
   const credentialTypeID = selectedCredentialTypeId || watchedCredentialTypeId;
 
   const isGalaxyCredential =
     !!credentialTypes && credentialTypes?.[credentialTypeID]?.kind === 'galaxy';
 
+  const watchedRequiredFields = useWatch<{
+    name: string;
+    credential_type: number;
+  }>({
+    name: ['name', 'credential_type'],
```

Comment:
Did I misread this one? Or are we hardcoding those values for all credential types, or do we need to take in consideration the dynamic required fields as well?